nemo/nemo/core/neural_modules.py

```diff
@@ -81,12 +83,13 @@ def __init__(
         #        "arguments:".format(self.__class__.__name__))
         #    nemo.logging.warning("{}".format(kwargs.keys()))
 
+    @deprecated()
     @staticmethod
     def create_ports(**kwargs):
         """ Deprecated method, to be remoted in the next release."""
         raise Exception(
-            'Deprecated method. Please implement inputs and outputs \
-                 properties to define module ports')
+            'Deprecated method. Please implement ``inputs`` and ``outputs`` \
```

Comment:
This particular case makes me think--should an "additional note" argument be in the deprecated decorator (maybe instead of or in addition to alternative_function)?

It seems to me that in the case of multiple alternative functions or other future deprecations we may want to provide some explanation or additional guidance than just an alternative function name. And since we have the decorator... two birds one stone, rather than having both a decorator and an additional logger warning/exception. But it may make it bulky.

What do you think?

Reply:
Yeah, actually I have faced similar case with create_ports() - we want to use two methods instead of one;)

How to call it: "explanation"?

Reply:
done! :+1: